packages/at_client/lib/src/client/at_client_impl.dart

```diff
@@ -967,4 +978,40 @@ class AtClientImpl implements AtClient {
     _preference = preference;
     _namespace = namespace;
   }
+
+  @override
+
+  /// Inject an instance of [AtChops]
+  /// if [AtChops] is not injected then pkam and encryption keys should be saved to local secondary before invoking this method.
+  void setAtChops({AtChops? atChops}) async {
```

Comment:
Looks like this method is only called from one place in the codebase in at_client_service.dart, and when it is called from that place, it is called with an AtChopsImpl

So a few questions
(1) When do you envisage the fallback code here being executed? - i.e. when would we be calling setAtChops(null) ?
(2) Can we move the async code in this method somewhere else? i.e. can this be a standard setter?

Reply:
#1 atChops object is mandatory for an at client instance because without at_chops encryption/decryption/data signing cannot be performed.
i put the fallback code in case the caller doesn't pass an instance of at_chops. There may be other packages apart from at_client_mobile that can create an instance of at client and call get/put etc.,
#2 Will think about a solution for this.

Reply:
I already had the same idea and tried the change earlier today. When we are in the init method, there are two scenarios
1) local secondary is not available when isLocalStoreRequired=false
2) local secondary is available but encryption/pkam keys are not yet persisted.
For both 1 and 2, we will not be able to create an instance of at chops. in the latest commit, I changed the setter to dart convention and moved the async behavior to a new private method(currently not called)
I will uptake at chops in a package that uses at client without the onboarding flow and explore the options to create at chops instance.